### PR TITLE
Base64 data issue/v5

### PR DESCRIPTION
**Pull request by inashivb**
Previous PR: #8651

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW1_files_sha256.
 | field | baseline | test | % |
|--|--|--|--| 
| | TREX_GENERIC_stats_chk | | |
| .capture.kernel_drops | 0 | 68 | 0.00 | 

Pipeline 12924

File: src/util-base64.c [hunk 1]
```diff
@@ -156,11 +156,27 @@ Base64Ecode DecodeBase64(uint8_t *dest, uint32_t dest_size, const uint8_t *src,
...
+        uint32_t numDecoded_blk = ASCII_BLOCK - (padding < B64_BLOCK ? padding : ASCII_BLOCK);
+        if (dest_size < *decoded_bytes + numDecoded_blk) {
+            SCLogDebug("Destination buffer full");
+            ecode = BASE64_ECODE_BUF;
+            *consumed_bytes += leading_sp;
+            return ecode;
+        }
+        /* if the destination size is not at least 3 Bytes long, it'll give a dynamic
+         * buffer overflow while decoding, so, return and let the caller take care of the
+         * remaining bytes to be decoded which should always be < 4 at this stage */
```
Comment (jlucovsky) [inline src/util-base64.c:172]: this might be a nit -- the comment says "< 4" but the code has `< 3` ... should the code be `<= 3`?

Comment (inashivb) [reply src/util-base64.c:172]: Not really. But, the comment could be clearer. 3 Bytes are the decoded bytes that need to fit in the destination buffer. < 4 Bytes comment is for the bytes to be decoded..

**Comment by inashivb** [comment]
Needs rebase and fixes.

**Pull request closed without merging**